executor/joiner.go

```diff
@@ -259,9 +260,14 @@ func (j *baseJoiner) filter(
 		innerColOffset, outerColOffset = 0, len(lUsed)
 		if !j.outerIsRight {
 			innerColOffset, outerColOffset = len(lUsed), 0
+			// the same outers are pruned, remaining the different inners
+			if innerColOffset-outerColOffset == 0 {
+				prunedOuters = true
+			}
 		}
+
 	}
-	return chunk.CopySelectedJoinRowsWithSameOuterRows(input, innerColOffset, outerColOffset, j.selected, output)
+	return chunk.CopySelectedJoinRowsWithSameOuterRows(input, innerColOffset, outerColOffset, j.selected, output, prunedOuters)
```

Comment:
`prunedOuters` seems to be confused to readers. How about explicitly send `innerColLen` and `outerColLen` as arguments.

Reply:
This corner case happens when:
(1) j.outerIsRight is false
(2) innerColOff and outerColOff are 0
These two factors will prune the same outer parts. This result cannot be achieved by taking `innerColLen` and `outerColLen`. Maintaining these two parameters are more complex than only one parameter. The `prunedOuters` specifies the special case when the joined tuple without outer parts.